packages/web/src/scss/components/FileUploader/_FileUploaderInput.scss

```diff
@@ -85,10 +86,54 @@ $_field-name: 'FileUploaderInput';
     display: initial;
 }
 
+:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled)
+    :is(.FileUploaderInput__message, [data-element='validator_message']) {
+    @include form-fields-tools.message-disabled();
+}
+
+:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled) :is(.FileUploaderInput__helperText) {
+    @include form-fields-tools.helper-text-disabled();
+}
+
+// 3.
+// stylelint-disable selector-max-compound-selectors, selector-max-specificity
+:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled)
+    > .FileUploaderInput__input
+    ~ .FileUploaderInput__dropZone
+    > svg,
+:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled)
+    > .FileUploaderInput__input
+    ~ .FileUploaderInput__dropZone
+    > .FileUploaderInput__dropZoneLabel,
+:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled)
+    > .FileUploaderInput__input
+    ~ .FileUploaderInput__dropZone
+    > .FileUploaderInput__dropZoneLabel
+    > .FileUploaderInput__link {
+    @include form-fields-tools.input-disabled();
+}
+// stylelint-enable selector-max-compound-selectors, selector-max-specificity
+
 .FileUploaderInput .FileUploaderInput__dropZone:hover {
     border-color: theme.$input-drop-zone-border-color-hover;
 }
 
+// 3.
+// stylelint-disable selector-max-compound-selectors, selector-max-specificity
+.FileUploaderInput .FileUploaderInput__input:disabled ~ .FileUploaderInput__dropZone > svg,
+.FileUploaderInput
+    .FileUploaderInput__input:disabled
+    ~ .FileUploaderInput__dropZone
+    > .FileUploaderInput__dropZoneLabel,
+.FileUploaderInput
+    .FileUploaderInput__input:disabled
+    ~ .FileUploaderInput__dropZone
+    > .FileUploaderInput__dropZoneLabel
+    > .FileUploaderInput__link {
```

Comment:
If I read it correctly, you are styling two scenarios here:

1. `:is(.FileUploaderInput--disabled, .FileUploaderInput.is-disabled)`, and
2. `.FileUploaderInput .FileUploaderInput__input:disabled`.

- Both are using the same mixin and could be merged, is it correct?
- More importantly, why are there two sets of selectors anyway? I don't think we have a use case for having a disabled `input` inside a not-disabled root element, or do we?